src/services/porssisahko.rs

```diff
@@ -153,52 +173,100 @@ pub async fn get_price_chart() -> Result<Vec<u8>> {
-
-        ctx.draw_series([
-            // Grey out past prices
+        // Highlight the step segment that corresponds to the current time in red,
+        // and annotate it above the line with a small gray connector.
+        if let Some(cur_hp) = prices.iter().rev().find(|hp| Utc::now() > hp.start_date) {
+            let seg_start = cur_hp.start_date.with_timezone(&TZ);
+            let seg_end = (cur_hp.start_date + Duration::minutes(18)).with_timezone(&TZ);
+            let seg_price = cur_hp.price;
+
+            // Draw the red step segment for the current interval
+            ctx.draw_series(LineSeries::new(
+                vec![(seg_start, seg_price), (seg_end, seg_price)].into_iter(),
+                ShapeStyle::from(&RED).stroke_width(2),
+            ))?;
+
+            // Annotation position: a bit to the right of the segment and above the line
+            let text_offset = Duration::minutes(20);
+            let y_offset_val = (max_price - min_price) * 0.06_f32; // vertical offset for the annotation
+            let label_pos = (seg_end + text_offset, seg_price + y_offset_val);
+
+            // Draw a thin gray connector from the step (midpoint) up to the annotation
+            let seg_mid = seg_start + Duration::seconds((seg_end - seg_start).num_seconds() / 2);
+            ctx.draw_series(LineSeries::new(
+                vec![
+                    (seg_mid, seg_price + y_offset_val / 10.0),
+                    (seg_mid, seg_price + y_offset_val),
+                ]
+                .into_iter(),
+                BLACK.mix(0.3).filled().stroke_width(1),
+            ))?;
+
+            // Draw the annotation text above the connector
+            let cur_label = format!("{seg_price:.2}");
+            let cur_label_style = style::TextStyle::from(("sans-serif", 26).into_font())
+                .pos(Pos::new(HPos::Center, VPos::Bottom)); // bottom anchor -> text sits above the coord
+            ctx.draw_series(std::iter::once(Text::new(
+                cur_label,
+                label_pos,
+                cur_label_style,
+            )))?;
+        }
+
+        ctx.draw_series(std::iter::once(
+            // Grey out past price
             Rectangle::new(
                 [
                     (start_date, f32::NEG_INFINITY),
-                    (current_date - cur_price_line_thickness, f32::INFINITY),
-                ],
-                BLACK.mix(0.1).filled(),
-            ),
-            // Draw a line at current price
-            Rectangle::new(
-                [
-                    (current_date - cur_price_line_thickness, f32::NEG_INFINITY),
-                    (current_date + cur_price_line_thickness, f32::INFINITY),
+                    (current_date, f32::INFINITY),
                 ],
-                RED.mix(0.5).filled(),
+                BLACK.mix(0.08).filled(),
             ),
-        ])?;
-
-        // Draw some extra ticks on the x-axis for each hour.
```

Comment:
This was a nice feature, oof.

Reply:
Better? 
<img width="1024" height="768" alt="porssisahko_test" src="https://github.com/user-attachments/assets/89127f63-8cd3-4a91-856a-98f96b57191e" />